src/Options/Generic.hs

```diff
@@ -940,15 +940,33 @@ parseRecordWithModifiers
 parseRecordWithModifiers mods = fmap GHC.Generics.to (genericParseRecord mods)
 
 -- | Marshal any value that implements `ParseRecord` from the command line
+--
+-- If you need to modify the top-level 'ParserInfo' or 'ParserPrefs'
+-- use the 'getRecord'' function.
 getRecord
     :: (MonadIO io, ParseRecord a)
     => Text
     -- ^ Program description
     -> io a
-getRecord desc = liftIO (Options.customExecParser defaultParserPrefs info)
+getRecord desc = getRecord' header mempty
   where
     header = Options.header (Data.Text.unpack desc)
-    info = Options.info parseRecord header
+
+-- | Marshal any value that implements `ParseRecord` from the command line
+--
+-- This is the lower-level sibling of 'getRecord' and lets you modify
+-- the 'ParserInfo' and 'ParserPrefs' records.
+getRecord'
+    :: (MonadIO io, ParseRecord a)
+    => Options.InfoMod a
+    -- ^ 'ParserInfo' modifiers
+    -> Options.PrefsMod
+    -- ^ 'ParserPrefs' modifiers
+    -> io a
+getRecord' infoMods prefMods = liftIO (Options.customExecParser prefs info)
+  where
+    prefs  = Options.prefs (prefMods <> defaultParserPrefs)
```

Comment:
Should this be `defaultParserPrefs <> prefMods`?  I assume you want the user-supplied preferences to override the default preferences